clio-util/src/main/scala/org/broadinstitute/clio/util/AuthUtil.scala

```diff
@@ -0,0 +1,56 @@
+package org.broadinstitute.clio.util
+
+import java.nio.file.Path
+
+import com.google.auth.oauth2.GoogleCredentials
+import io.circe.parser.decode
+import org.broadinstitute.clio.util.json.ModelAutoDerivation
+import org.broadinstitute.clio.util.model.ServiceAccount
+import scala.io.Source
+
+object AuthUtil extends ModelAutoDerivation {
+
+  /** Scopes needed from Google to get past Clio's auth proxy. */
+  private val authScopes = Seq(
+    "https://www.googleapis.com/auth/userinfo.profile",
+    "https://www.googleapis.com/auth/userinfo.email"
+  )
+
+  /**
+    * Get google credentials either from the service account json or
+    * from the user's default setup.
+    * @param serviceAccountPath Option of path to the service account json
+    * @return
+    */
+  def getGoogleCredentials(
+    serviceAccountPath: Option[Path]
+  ): GoogleCredentials = {
+    serviceAccountPath
+      .map { jsonPath =>
+        loadServiceAccountJson(jsonPath)
+      }
+      .map(getCredsFromServiceAccount)
+      .orElse {
+        Option(GoogleCredentials.getApplicationDefault)
```

Comment:
Comment for the record: `GoogleCredentials.getApplicationDefault` apparently throws an exception if not run on a gcloud VM.